clumper/clump.py

```diff
@@ -174,19 +185,112 @@ def read_csv(cls, path, delimiter=",", fieldnames=None, n=None):
             if n <= 0:
                 raise ValueError("Number of lines to read must be > 0.")
 
-        if path.startswith("https:") or path.startswith("http:"):
+        # conveniently excludes pathlib files here
+        # and removes the need to write code to check pathlib files in other places
+        if isinstance(path, str) and path.startswith(("https:", "http:")):
             with urllib.request.urlopen(path) as resp:
                 if fieldnames is None:
                     fieldnames = resp.readline().decode().strip().split(",")
-                # this section allows us to chunk the rows, if nrows is supplied
+                # This section allows us to chunk the rows, if nrows is supplied
+                # one feature that should be added is null values representation
+                # not complex though, if it is an empty string or NA, then it should
+                # read as None. I'll try to do that in a future PR. Baby steps.
                 body = it.islice(resp, 0, n)
                 body = (word.decode().strip().split(",") for word in body)
                 body = it.product([fieldnames], body)
-                return Clumper([dict(zip(key, values)) for key, values in body])
+                result = [dict(zip(key, values)) for key, values in body]
+        else:
+            with open(path, newline="") as csvfile:
+                reader = csv.DictReader(
+                    csvfile, delimiter=delimiter, fieldnames=fieldnames
+                )
+                reader = csv.DictReader(
+                    csvfile, delimiter=delimiter, fieldnames=fieldnames
+                )
```

Comment:
This feels like like repetition.

Reply:
Thanks for this. Missed it. I'll have a look at it today